imsv-docs-astro/src/models/ContentRegistry.mjs

```diff
@@ -0,0 +1,167 @@
+  #tokensByName = {};
+
+  /** @type {Object<string,SupportedChain>} */
+  #chainsByName = {};
+
+  /** @type {Object<string,SupportedNetwork>} */
+  #networksByName = {};
+
+  /** @type {Object<string,FundingType>} */
+  #fundingTypesByName = {};
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerToken(content) {
+    const token = SupportedToken.fromContent(content);
+    this.#tokensByName[token.name] = token;
+  }
+
+  getToken(name) {
+    if (!this.#tokensByName[name]) {
+      throw Error(`Unregistered token: ${name}`);
+    }
+    return this.#tokensByName[name];
+  }
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerFundingProtocol(content) {
+    const protocol = FundingProtocol.fromContent(content);
+    this.#protocolsByName[protocol.name] = protocol;
+  }
+
+  getFundingProtocol(name) {
+    if (!this.#protocolsByName[name]) {
+      throw Error(`Unregistered protocol: ${name}`);
+    }
+    return this.#protocolsByName[name];
+  }
+
+  /**
+   * @param {Object} opts
+   * @param {string} opts.networkName
+   * @param {string} opts.protocolName
+   * @returns {DeployedFundingProtocol}
+   * */
+  getDeployedFundingProtocol({ networkName, protocolName }) {
+    return this.getFundingProtocol(protocolName).getDeployedInstance(networkName);
+  }
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerChain(content) {
+    const chain = SupportedChain.fromContent(content);
+    this.#chainsByName[chain.name] = chain;
+  }
+
+  getChain(name) {
+    if (!this.#chainsByName[name]) {
+      throw Error(`Unregistered chain: ${name}`);
+    }
+    return this.#chainsByName[name];
+  }
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerNetwork(content) {
+    const network = SupportedNetwork.fromContent({ registry: this, content });
+    this.#networksByName[network.name] = network;
+  }
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerNetworkToken(content) {
+    NetworkToken.fromContent({ registry: this, content });
+  }
+
+  /**
+   * @param {Object} opts
+   * @param {string} opts.tokenName
+   * @param {string} opts.networkName
+   * @returns {NetworkToken}
+   */
+  getNetworkToken({ tokenName, networkName }) {
+    return this.getToken(tokenName).getInstance(networkName);
+  }
+
+  getNetwork(name) {
+    if (!this.#networksByName[name]) {
+      throw Error(`Unregistered network: ${name}`);
+    }
+    return this.#networksByName[name];
+  }
+
+  /**
+   * @param {CollectionEntry} content
+   */
+  registerFundingType(content) {
+    const fundingType = FundingType.fromContent({ registry: this, content });
+    this.#fundingTypesByName[fundingType.name] = fundingType;
+  }
+
+  getFundingType(name) {
+    if (!this.#fundingTypesByName[name]) {
+      throw Error(`Unregistered funding type: ${name}`);
+    }
+    return this.#fundingTypesByName[name];
+  }
+
+  static async create() {
+    const registry = new ContentRegistry();
```

Comment:
Style preference. Spacing between the registry entry creation calls would make it more readable.